src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapperTest.java

```diff
@@ -458,6 +461,128 @@ void should_throw_exception_when_security_classification_enum_is_not_mapped() {
+    }
+
+    @Test
+    void should_map_task_resource_to_task_when_work_type_is_null() {
+        ZonedDateTime createdDate = ZonedDateTime.now();
+        String formattedCreatedDate = CAMUNDA_DATA_TIME_FORMATTER.format(createdDate);
+        ZonedDateTime dueDate = createdDate.plusDays(1);
+        String formattedDueDate = CAMUNDA_DATA_TIME_FORMATTER.format(dueDate);
+
+        List<TaskAttribute> attributes = getDefaultAttributes(formattedCreatedDate, formattedDueDate);
+
+        TaskResource taskResource = cftTaskMapper.mapToTaskResource(taskId, attributes);
+        Task task = cftTaskMapper.mapToTask(taskResource);
+
+        assertNull(task.getWorkTypeId());
+
+        assertNull(taskResource.getWorkTypeResource());
+    }
+
+    @Test
+    void should_map_task_resource_to_task_when_work_type_not_exists() {
+        ZonedDateTime createdDate = ZonedDateTime.now();
+        String formattedCreatedDate = CAMUNDA_DATA_TIME_FORMATTER.format(createdDate);
+        ZonedDateTime dueDate = createdDate.plusDays(1);
+        String formattedDueDate = CAMUNDA_DATA_TIME_FORMATTER.format(dueDate);
+
+        List<TaskAttribute> attributes =
+            getDefaultAttributesWithoutWithWorkType(formattedCreatedDate, formattedDueDate);
+
+        TaskResource taskResource = cftTaskMapper.mapToTaskResource(taskId, attributes);
+        taskResource.setWorkTypeResource(null);
+
+        Task task = cftTaskMapper.mapToTask(taskResource);
+
+        assertNull(task.getWorkTypeId());
+
+        AssertionsForClassTypes.assertThatThrownBy(() -> taskResource.getWorkTypeResource().getId())
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void should_throw_exception_when_map_task_resource_to_task_and_due_date_is_null() {
+        List<TaskAttribute> attributes = getDefaultAttributesWithoutDueDate();
+
+        AssertionsForClassTypes.assertThatThrownBy(() -> cftTaskMapper.mapToTaskResource(taskId, attributes))
+            .isInstanceOf(NullPointerException.class)
+            .hasNoCause()
+            .hasMessage("TASK_DUE_DATE must not be null");
+    }
+
+    @Test
+    void should_map_task_resource_to_task_when_create_date_is_null() {
+        String formattedCreatedDate = null;
+        ZonedDateTime dueDate = ZonedDateTime.now().plusDays(1);
+        String formattedDueDate = CAMUNDA_DATA_TIME_FORMATTER.format(dueDate);
+
+        List<TaskAttribute> attributes = getDefaultAttributes(formattedCreatedDate, formattedDueDate);
+
+        TaskResource taskResource = cftTaskMapper.mapToTaskResource(taskId, attributes);
+        assertNull(taskResource.getCreated());
+        assertNotNull(taskResource.getDueDateTime());
+
+        assertThatThrownBy(() -> cftTaskMapper.mapToTask(taskResource))
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void should_throw_exception_map_task_resource_to_task_when_due_date_is_null() {
```

Comment:
Is this test already done in? should_throw_exception_when_map_task_resource_to_task_and_due_date_is_null?

Reply:
I changed the name of the methods